packages/expo-doctor/src/doctor.ts

```diff
@@ -0,0 +1,118 @@
+import { getConfig } from '@expo/config';
+import chalk from 'chalk';
+import semver from 'semver';
+
+import { ExpoConfigSchemaCheck } from './checks/ExpoConfigSchemaCheck';
+import { GlobalPrereqsVersionCheck } from './checks/GlobalPrereqsVersionCheck';
+import { IllegalPackageCheck } from './checks/IllegalPackageCheck';
+import { InstalledDependencyVersionCheck } from './checks/InstalledDependencyVersionCheck';
+import { SupportPackageVersionCheck } from './checks/SupportPackageVersionCheck';
+import { DoctorCheck, DoctorCheckParams } from './checks/checks.types';
+import { Log } from './utils/log';
+import { logNewSection } from './utils/ora';
+import { ltSdkVersion } from './utils/versions';
+import { warnUponCmdExe } from './warnings/windows';
+
+export async function runCheckAsync(
+  check: DoctorCheck,
+  checkParams: DoctorCheckParams
+): Promise<boolean> {
+  // bail if check isn't relevant for SDK version
+  if (
+    checkParams.exp.sdkVersion !== 'UNVERSIONED' &&
+    !semver.satisfies(checkParams.exp.sdkVersion!, check.sdkVersionRange)
+  ) {
+    return true;
+  }
+
+  const ora = logNewSection(check.description);
+  let result;
+  try {
+    result = await check.runAsync(checkParams);
+  } catch (e: any) {
+    if (e.code === 'ENOTFOUND') {
+      Log.error(
+        `Error: this check requires a connection to the Expo API. Please check your network connection.`
+      );
+    } else {
+      Log.exception(e);
+    }
+    ora.fail();
+    return false;
+  }
+  if (result.isSuccessful) {
+    ora.succeed();
+    return true;
+  }
+  ora.fail();
+  if (result.issues.length) {
+    Log.log(chalk.underline(chalk.yellow(`Issues:`)));
```

Comment:
```suggestion
    Log.log(chalk.underline.yellow(`Issues:`));
```
I think these can be chained btw. ndb